app/controllers/api/physical_servers_controller.rb

```diff
@@ -80,6 +78,18 @@ def refresh_physical_server(physical_server)
       action_result(false, err.to_s)
     end
 
+    def ensure_id_present(type, id)
+      raise BadRequestError, "Must specify an id for changing a #{type} resource" unless id
+    end
+
+    def ensure_resource_exists(type, id)
+      raise ActiveRecord::RecordNotFound if single_resource? && !collection_class(type).exists?(id)
```

Comment:
why is this method needed ?  resource_search would throw the NotFoundError plus do the necessary RBAC validation.

Reply:
@abellotti it does, but we don't want it caught by the `api_action` block, which will always respond with a 200. Performing another `resource_search` would be unnecessarily expensive if we only want to check for existence.

@douglasgabriel you may want to use our own `NotFoundError` here